packages/datastore/src/table.ts

```diff
@@ -416,18 +454,20 @@ namespace Private {
   };
 
   /**
-   * Apply a patch to a record.
+   * Apply or unapply a patch to a record.
    *
    * @param schema - The schema for the record.
    *
    * @param record - The record of interest.
    *
    * @param patch - The patch to apply to the record.
    *
+   * @param which - whether to apply or unapply the patch.
+   *
    * @return The result of applying the patch.
    */
   export
-  function applyPatch<S extends Schema>(schema: S, record: Record<S>, patch: Record.Patch<S>): PatchResult<S> {
+  function patch<S extends Schema>(schema: S, record: Record<S>, patch: Record.Patch<S>, which: 'apply' | 'unapply'): PatchResult<S> {
```

Comment:
I think I would make this two separate functions, it doesn't look like we simplify much using the `which` arg.

Reply:
Okay, sounds good.